### PR TITLE
fix(auth): handleMpCancel remove on BA id match [WIP]

### PR DESCRIPTION
**Pull request by StaberindeZA**
## Because

- Should a user get a new billing agreement added to their customer
  record, while a prior one was recently deleted, but its IPN
  notification arrives later, we will remove the newer billing agreement
  ID from the Stripe customer metadata.

## This pull request

- Removes the PayPal billing agreement ID from Stripe only if the
  billing agreement ID from the IPN matches what is in Stripe. If the
  IDs do not match, then only the database billing agreement should be
  updated.

## Issue that this pull request solves

Closes: #9328

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

File: packages/fxa-auth-server/lib/routes/subscriptions/paypal-notifications.ts [hunk 1]
```diff
@@ -80,10 +86,39 @@ export class PayPalNotificationHandler extends PayPalHandler {
...
+   * If the Billing Agreement ID from the IPN request does not match the
+   * ID on the Stripe customer, then only update the database.
+   */
+  private async removeBillingAgreement(
+    customer: Stripe.Customer,
+    billingAgreement: PayPalBillingAgreements,
+    account: Account
+  ) {
+    if (
+      customer?.metadata[STRIPE_CUSTOMER_METADATA.PAYPAL_AGREEMENT] ===
```
Comment (bbangert) [inline packages/fxa-auth-server/lib/routes/subscriptions/paypal-notifications.ts:100]: I think we have a helper function to get this.

File: packages/fxa-auth-server/lib/routes/subscriptions/paypal-notifications.ts [hunk 2]
```diff
@@ -80,10 +86,39 @@ export class PayPalNotificationHandler extends PayPalHandler {
...
+      billingAgreement.billingAgreementId
+    ) {
+      await this.stripeHelper.removeCustomerPaypalAgreement(
+        account.uid,
+        customer.id,
+        billingAgreement.billingAgreementId
+      );
+    } else {
+      updatePayPalBA(
```
Comment (bbangert) [inline packages/fxa-auth-server/lib/routes/subscriptions/paypal-notifications.ts:109]: No await here?

Comment (StaberindeZA) [reply packages/fxa-auth-server/lib/routes/subscriptions/paypal-notifications.ts:109]: Ah yes, missed this, thank you.

File: packages/fxa-auth-server/lib/routes/subscriptions/paypal-notifications.ts (hunk 1 above)
Comment (StaberindeZA) [reply packages/fxa-auth-server/lib/routes/subscriptions/paypal-notifications.ts:100]: There is! Thank you.

**Review by bbangert — APPROVED** [review]
LGTM! Why a draft? 😉

**Comment by StaberindeZA** [comment]
> LGTM! Why a draft? 😉

Thank you, thank you. I wanted to get some local testing done before merging, which is why it was in draft.

**Pull request merged**